### ISSUE TITLE
Implement SAPS

### ISSUE THREAD

**Issue by ilknur12** [Cortys/SEP-Probly-WS2526#521]
## Context
After the implementation of APS (Adaptive Prediction Sets) and RAPS, the next method to add is SAPS (Sorted Adaptive Prediction Sets) for conformal prediction, as defined in the referenced paper.

## Goal
Implement the method SAPS in the existing architecture following the same pattern as APS, LAC, and RAPS.

## Tasks
1. **Architecture Extension:**
   - [x] Create `src/probly/conformal_prediction/scores/saps/` with same structure as APS
   - [x] Add files: `__init__.py`, `common.py`, `flax.py`, `torch.py`

2. **Implement SAPS (`scores/saps/...`):**
   - [x] Implement saps_score_func using @lazydispatch
   - [x] Create SAPSScore class implementing the Score protocol
   - [x] Implement adaptive scoring mechanism based on equation 10 from the paper:
```
 S_saps(x, y, u) = {
    u * π_max(x)                   if o(y) = 1
    π_max(x) + (o(y) - 2 + u) * λ   otherwise }
```
where: 
- `π_max(x)` is the maximum softmax probability
- `o(y)` is the rank position of label y (1-based)
- `u` is an uniform random variable in [0,1)
- `λ` is a hyperparameter controlling the weight of ranking information

3. **Framework Support:**
   - [x] Add torch.py with PyTorch implementation 
   - [x] Add flax.py with JAX/Flax implementation of `saps_score_func`
   - [x] Register implementations in `__init__.py` with delayed registration

4. **Implementaion includes:**
   - [x] Core Idea: SAPS discards all softmax probability values except the maximum probability, replacing other values with a constant based on label ranking
   - [x] Key Insight: Probability values beyond the maximum are redundant for conformal prediction; using only ranking information produces more compact prediction sets
   - [x] Compatibility: Must work with existing SplitConformalPredictor
   - [x] Parameters: 
- `lambda_val (λ)`: Weight parameter for ranking information (Eq. 10)
- `random_state`: For reproducibility of randomization

## References
[Huang et al. (2023): "Conformal Prediction for Deep Classifier via Label Ranking"](https://arxiv.org/pdf/2310.06430)
